src/plugins/pcr/src/PrimerStatistics.cpp

```diff
@@ -34,7 +34,8 @@
 
 namespace U2 {
 
-bool PrimerStatistics::checkPcrPrimersPair(const QByteArray &forward, const QByteArray &reverse, QString &message) {
+bool PrimerStatistics::checkPcrPrimersPair(const QByteArray &forward, const QByteArray &reverse, QString &message, bool &isCriticalError) {
```

Comment:
You can return the error message as the method result. If it is empty there is no error.
This will remove duplicated state: return value (true|false) and the error message state (empty/not empty)